src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/TypeRef.DefaultResolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This is consumed as 'generated' code in a source package and therefore requires an explicit nullable enable
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CommonLanguageServerProtocol.Framework;
+
+internal readonly partial record struct TypeRef
+{
+    private sealed class DefaultResolverImpl : ITypeRefResolver
+    {
+        private static readonly Dictionary<string, Type> s_typeNameToTypeMap = [];
+
+        public Type? Resolve(TypeRef typeRef)
+        {
+            if (typeRef.IsDefault)
+            {
+                return null;
+            }
+
+            var typeName = typeRef.TypeName;
+
+            lock (s_typeNameToTypeMap)
```

Comment:
wondering if there is any value in using a concurrent dictionary (or a read first outside of the lock).  since this will get hit on every request

Reply:
and any reason we don't do the same caching in the ExtensionTypeRefResolver?  If it would be useful there wondering if `ITypeRefResolver` should be an abstract base that does the caching for all impls.

Reply:
An abstract base class is a good idea. Will do that.